### PR TITLE
feat: implement executeWorkflow (V2) unit test

### PR DESCRIPTION
**Pull request by rhkp**
Implement executeWorkflow (V2) unit test
Implement mapToExecuteWorkflowResponseDTO unit test

**Implemented Test Result**
image

### CODE REVIEW

File: plugins/orchestrator-backend/src/service/api/mapping/V2Mappings.test.ts [hunk 1]
```diff
@@ -1,11 +1,35 @@
...
+    const execWorkflowResp = {
+      id: uuidv4(),
+    };
+    const workflowId = uuidv4();
+    const mappedValue = mapToExecuteWorkflowResponseDTO(
+      workflowId,
+      execWorkflowResp,
+    );
+    expect(mappedValue).toBeDefined();
+    expect(mappedValue.id).toBeDefined();
```
Comment (gciavarrini) [inline plugins/orchestrator-backend/src/service/api/mapping/V2Mappings.test.ts:23]: In addition to these assertions, do you think can be a good idea to add a check on the number of properties available in `ExecuteWorkflowResponseDTO`? So if in future we add a new property, the test will let us know to update it according.
Something like `expect(Object.keys(mappedValue).length).toBe(1);` 

WDYT?

File: plugins/orchestrator-backend/src/service/api/mapping/V2Mappings.test.ts [hunk 2]
```diff
@@ -1,11 +1,35 @@
...
+  });
+
+  it('throws error when no id attribute present in response', async () => {
+    expect(() => {
+      mapToExecuteWorkflowResponseDTO('workflowId', { id: '' });
+    }).toThrow();
```
Comment (gciavarrini) [inline plugins/orchestrator-backend/src/service/api/mapping/V2Mappings.test.ts:29]: Do you think it's a good idea to the the error message too?

File: plugins/orchestrator-backend/src/service/api/v2.test.ts [hunk 3]
```diff
@@ -25,6 +28,58 @@ jest.mock('../SonataFlowService', () => {
...
+    const result: ExecuteWorkflowResponseDTO = await V2.executeWorkflow(
+      mockDataIndexService,
+      mockSonataFlowService,
+      workflowData,
+      uuidv4(),
+      'businessKey',
+    );
+
+    // Assert
+    expect(result).toEqual(mockExecWorkflowResponseV1);
```
Comment (gciavarrini) [inline plugins/orchestrator-backend/src/service/api/v2.test.ts:80]: Can you double check this?
This asserts that v2 result is equal to mocked v1 result. Maybe it's better to decouple the two?

File: plugins/orchestrator-backend/src/service/api/mapping/V2Mappings.test.ts [hunk 4]
```diff
@@ -1,11 +1,35 @@
...
 
 import { generateTestWorkflowOverview } from '../test-utils';
 import {
+  mapToExecuteWorkflowResponseDTO,
   mapToWorkflowOverviewDTO,
   mapWorkflowCategoryDTOFromString,
 } from './V2Mappings';
 
+describe('scenarios to verify executeWorkflowResponseDTO', () => {
+  it('correctly maps positive scenario response', async () => {
```
[the remark this answers is not in the document]
Comment (rhkp) [reply plugins/orchestrator-backend/src/service/api/mapping/V2Mappings.test.ts:13]: I chose for separate describe for each endpoint so as to have linear structure in file. Theoretically we can nest describe/it within each other but that may make it difficult for following code. I recommend to follow linear structure. WDYT?

**Pull request closed without merging**